pkg/proto/types.go

```diff
@@ -295,7 +295,26 @@ type Order interface {
 	GetMatcherPK() crypto.PublicKey
 }
 
-type order struct {
+func OrderToOrderData(o Order) (OrderData, error) {
+	switch o.GetVersion() {
+	case 1:
+		o, ok := o.(OrderV1)
+		if !ok {
+			return OrderData{}, errors.New("failed to cast an order version 1 to OrderV1")
+		}
+		return o.OrderData, nil
+	case 2:
+		o, ok := o.(OrderV2)
+		if !ok {
+			return OrderData{}, errors.New("failed to cast an order version 2 to OrderV2")
+		}
+		return o.OrderData, nil
+	default:
+		return OrderData{}, errors.New("invalid order version")
+	}
+}
+
+type OrderData struct {
```

Comment:
I suggest to name this structure 'OrderBody', or may be we should rename Order interface...

Reply:
Done.